### PR TITLE
CC-684 Added Staging Terraform config vars

### PR DESCRIPTION
**Pull request by mwestacott**
https://companieshouse.atlassian.net/browse/CC-684

* Added config specific to the Staging environment

### CODE REVIEW

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 1]
```diff
@@ -1 +1,62 @@
-# placeholder file
\ No newline at end of file
+environment = "staging"
+aws_profile = "staging-eu-west-2"
+
+
+# TODO Check not required (not there in OE Web Staging config)
```
Comment (mwestacott) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:5]: A number of these TODOs in this file, if people could weigh in, would be good. Or we can have a call.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 2]
```diff
@@ -1 +1,62 @@
...
+
+# TODO Needed? Not present in Mesos Staging config and variable only defined in the terraform but not used anywhere in the code that I can see
+#
+# cookie_secret_flag = "0"
+
+
+# TODO Not present in Mesos Staging config (log_level isn't neither, but I think default value of 'debug' is being applied there,
+#      so making it explicit is good - that's what we did with OE Web). I just don't think human_log needs setting, if we agree? 
+#
+# human_log = "1"
```
Comment (markpit) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:52]: could be a CH library needs it for logging?

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 3]
```diff
@@ -1 +1,62 @@
...
+
+
+# TODO Needed? Present in Mesos Staging config, variable already defined in the terraform but not used anywhere in the code that I can see
+#
+# export AUTHENTICATION_MIDDLEWARE=on
```
Comment (markpit) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:57]: could be needed by imported libraries, if it is in mesos config i'd keep them in this config too

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 2 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:52]: Yes, I think the CH Node structured logging library does use it. Not set in Staging Mesos though. In Mesos CiDev env neither, so now wondering why it's here at all...

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 1 above)
Comment (tobafolaranmi-ch) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:5]: No they're not needed

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 4]
```diff
@@ -1 +1,62 @@
...
+
+
```
Comment (tobafolaranmi-ch) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:59]: If a param is defined in Mesos but not used in code, can we verify if it should be taken out completely

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 1 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:5]: 👍 thanks.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 5]
```diff
@@ -1 +1,62 @@
...
+max_extension_period_in_months = "12"
+log_level = "debug"
+cookie_name = "__SID"
+
+# TODO CiDev setting was 1200 but set to this value (3600) in Mesos Staging config
+default_session_expiration = "3600"
+
+# TODO Needed? Not present in Mesos Staging config and variable only defined in the terraform but not used anywhere in the code that I can see
+#
+# cookie_expiration_in_seconds = "3600"
```
[the remark this answers is not in the document]
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:41]: Could be, but if no definition in Mesos for CiDev or Staging (in both app_env and global_env) I don't see why it should be here. Should also go from the CiDev config, I believe.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 3 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:57]: Makes sense. It's not defined in the CiDev vars file but it is in the CiDev Mesos config too, so I think we missed that.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 4 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:59]: I think that the best we can do to check is search CH GitHub.

https://github.com/search?q=org%3Acompanieshouse+AUTHENTICATION_MIDDLEWARE&type=code&p=1

This suggests it's only used/defined by extensions web but there's no code that actually uses it. Maybe it was an old config value used in extensions web but the associated code got removed, with no clean up/removal of the variable.

Like I said, it's missing in the CiDev var file, so if we leave it like that and all testing passes in CiDev, it suggests it's no longer needed.

Comment (tobafolaranmi-ch) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:59]: Alright, let's do that then

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 3 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:57]: See comment below - looks like this isn't needed, so planning to remove it.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 1 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: Now removed. Resolving this comment.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 6]
```diff
@@ -1 +1,31 @@
...
+max_file_size_bytes = "4194304"
+show_service_unavailable_page = "off"
+too_soon_days_before_due_date = "60"
+max_extension_period_in_months = "12"
+log_level = "debug"
+cookie_name = "__SID"
+default_session_expiration = "3600"
+
+
+# TODO Needed? Present in Mesos Staging config but not CiDev config, variable not defined in the terraform but there's
```
Comment (mwestacott) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:28]: The only outstanding issue now (at least from my perspective). I don't think this is needed but what do others think?

Comment (markpit) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:28]: doesn't look like it is used so can go without

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 5 above)
[the remark this answers is not in the document]
Comment (markpit) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: ok, if not found in mesos app_env, global_env or private_env then should be safe to be left out

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 2 above)
Comment (markpit) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: ok can leave out

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 6 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: Removed, so resolving.

**Review by tobafolaranmi-ch — APPROVED** [review]
LGTM

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 5 above)
[the remark this answers is not in the document]
Comment (markpit) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: just to confirm, was it present in the private_env?

[the remark this answers is not in the document]
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars]: No. Not there in CiDev or Staging on Mesos.

**Review by markpit — APPROVED** [review]

**Pull request merged**